ChartboostMediationDemo/Demo_SwiftUI/QueuedAdView.swift

```diff
@@ -0,0 +1,138 @@
+    @EnvironmentObject var viewModel: AdQueueViewModel
+
+    var body: some View {
+        VStack {
+            Image("Queued")
+                .padding(.vertical, 32)
+
+            Text("Fullscreen ads (both Interstitial and Rewarded) can be pre-loaded in a FullscreenAdQueue")
+                .multilineTextAlignment(.center)
+                .padding(.horizontal, 32)
+
+            Button {
+                viewModel.toggleRunningState()
+            } label: {
+                Text(viewModel.runButtonText)
+                    .font(.title2)
+                    .foregroundColor(.white)
+                    .frame(height: 48)
+                    .frame(maxWidth: .infinity)
+            }
+            .buttonStyle(.borderedProminent)
+            .padding(.horizontal, 32)
+
+            HStack(spacing: 0) {
+                ForEach(0..<5) { index in
+                    Text(viewModel.numberOfAdsReady >= index + 1 ? "🟩" : "🔲")
+                        .font(.largeTitle)
+                        .frame(maxWidth: .infinity)
+                }
+            }
+
+            Text("As soon as there's at least one loaded ad in the queue, you can begin showing them.")
+                .multilineTextAlignment(.center)
+                .padding(.horizontal, 32)
+
+            if let topViewController = UIApplication.topViewController {
+                Button {
+                    if let ad = viewModel.getNextAd() {
+                        ad.show(with: topViewController) { result in
+                            if let error = result.error {
+                                print("[Error] showing fullscreen advertisement (name: \(error.chartboostMediationCode.name), code: \(error.code))")
+                            }
+                            else {
+                                print("[Success] did show fullscreen advertisement for placement")
+                            }
+                        }
+                    }
+                } label: {
+                    Text("Show")
+                        .font(.title2)
+                        .foregroundColor(.white)
+                        .frame(height: 48)
+                        .frame(maxWidth: .infinity)
+                }
+                .buttonStyle(.borderedProminent)
+                .padding(.horizontal, 32)
+                .disabled(viewModel.numberOfAdsReady == 0)
+            }
+
+            Spacer()
+        }
+        .onAppear {
+            viewModel.setDelegate()
+        }
+    }
+}
+
+// A FullscreenAdQueueDelegate can be used to receive updates about queue events.
+class QueueDelegate: ObservableObject, FullscreenAdQueueDelegate {
+    var viewModel: AdQueueViewModel
+
+    init(viewModel: AdQueueViewModel) {
+        self.viewModel = viewModel
+    }
+
+    func fullscreenAdQueue(_ adQueue: FullscreenAdQueue, didFinishLoadingWithResult: ChartboostMediationAdLoadResult, numberOfAdsReady: Int) {
+        DispatchQueue.main.async {
```

Comment:
These dispatches should be unnecessary since we always make sure to call delegate methods on the main queue in the SDK right?